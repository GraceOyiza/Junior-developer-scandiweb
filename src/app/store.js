import { configureStore } from '@reduxjs/toolkit';
import cartReducer from '../feature/cartSlice';
import currencyReducer from '../feature/currencySlice';
import productReducer from '../feature/productSlice';
import categoryReducer from '../feature/categorySlice';

export default configureStore({
  reducer: {
    cart: cartReducer,
    currency: currencyReducer,
    product: productReducer,
    category: categoryReducer,
  },
});