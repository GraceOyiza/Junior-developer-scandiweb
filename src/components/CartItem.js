import React from 'react';



class CartItem extends React.Component {
 

  render() {
    return (
      <div>
        <Flex>
          <div>
            <Pag> {this.props.cartItem.name}</Pag>
            <Pag> {this.props.cartItem.brand}</Pag>
            <Price>
              {this.props.cartItem.prices[0].currency.symbol}
              {this.props.cartItem.prices[0].amount}
            </Price>
            {this.props.cartItem.attributes.map((attribute, index) => (
              <div key={index}>
                <VariantList
                  title={attribute.name}
                  type={attribute.type}
                  variants={attribute.items}
                  selectVariant={this.props.selectVariant}
                />
              </div>
            ))}
          </div>
          <Flex>
            <Flex1>
              <Button
                onClick={() =>
                  this.props.incrementQuantity(this.props.cartItem.id)
                }
              >
                <svg
                  width="24"
                  height="24"
                  viewBox="0 0 24 24"
                  fill="none"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 5V19"
                    strokeWidth="#32323A"
                    strokeWidth-width="2"
                    strokeWidthLinecap="round"
                    strokeWidth-linejoin="round"
                  />
                  <path
                    d="M5 12H19"
                    strokeWidth="#32323A"
                    strokeWidth-width="2"
                    strokeWidthLinecap="round"
                    strokeWidth-linejoin="round"
                  />
                </svg>
              </Button>
              <QuantityInput value={this.props.cartItem.quantity} />
              <Button
                onClick={() =>
                  this.props.decrementQuantity(this.props.cartItem.id)
                }
              >
                <svg
                  width="16"
                  height="2"
                  viewBox="0 0 16 2"
                  fill="none"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1 1H15"
                    strokeWidth="#32323A"
                    strokeWidth-width="2"
                    strokeWidthLinecap="round"
                    strokeWidth-linejoin="round"
                  />
                </svg>
              </Button>
            </Flex1>

            <div>
              {this.props.cartItem.gallery[0].gallery}
              <Image src={this.props.cartItem.gallery[0]} />
            </div>
          </Flex>
        </Flex>
      </div>
    );
  }
}

export default CartItem;
