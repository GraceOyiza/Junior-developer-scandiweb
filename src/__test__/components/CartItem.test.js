import React from 'react';
import Enzyme, { render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import store from '../../app/store';
import { BrowserRouter } from 'react-router-dom';
import CartItem from '../../components/CartItem';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let cartItem = {
  __typename: 'Product',
  id: 'ps-5',
  name: 'PlayStation 5',
  description:
    '<p>A good gaming console. Plays games of PS4! Enjoy if you can buy it mwahahahaha</p>',
  inStock: false,
  gallery: [
    'https://images-na.ssl-images-amazon.com/images/I/510VSJ9mWDL._SL1262_.jpg',
    'https://images-na.ssl-images-amazon.com/images/I/610%2B69ZsKCL._SL1500_.jpg',
    'https://images-na.ssl-images-amazon.com/images/I/51iPoFwQT3L._SL1230_.jpg',
    'https://images-na.ssl-images-amazon.com/images/I/61qbqFcvoNL._SL1500_.jpg',
    'https://images-na.ssl-images-amazon.com/images/I/51HCjA3rqYL._SL1230_.jpg',
  ],
  brand: 'Sony',
  prices: [
    {
      __typename: 'Price',
      currency: {
        __typename: 'Currency',
        label: 'USD',
        symbol: '$',
      },
      amount: 844.02,
    },
    {
      __typename: 'Price',
      currency: {
        __typename: 'Currency',
        label: 'GBP',
        symbol: '£',
      },
      amount: 606.67,
    },
    {
      __typename: 'Price',
      currency: {
        __typename: 'Currency',
        label: 'AUD',
        symbol: 'A$',
      },
      amount: 1088.79,
    },
    {
      __typename: 'Price',
      currency: {
        __typename: 'Currency',
        label: 'JPY',
        symbol: '¥',
      },
      amount: 91147.25,
    },
    {
      __typename: 'Price',
      currency: {
        __typename: 'Currency',
        label: 'RUB',
        symbol: '₽',
      },
      amount: 63826.91,
    },
  ],
  attributes: [
    {
      __typename: 'AttributeSet',
      id: 'Color',
      name: 'Color',
      type: 'swatch',
      items: [
        {
          __typename: 'Attribute',
          displayValue: 'Green',
          value: '#44FF03',
          id: 'Green',
        },
        {
          __typename: 'Attribute',
          displayValue: 'Cyan',
          value: '#03FFF7',
          id: 'Cyan',
        },
        {
          __typename: 'Attribute',
          displayValue: 'Blue',
          value: '#030BFF',
          id: 'Blue',
        },
        {
          __typename: 'Attribute',
          displayValue: 'Black',
          value: '#000000',
          id: 'Black',
        },
        {
          __typename: 'Attribute',
          displayValue: 'White',
          value: '#FFFFFF',
          id: 'White',
        },
      ],
    },
    {
      __typename: 'AttributeSet',
      id: 'Capacity',
      name: 'Capacity',
      type: 'text',
      items: [
        {
          __typename: 'Attribute',
          displayValue: '512G',
          value: '512G',
          id: '512G',
        },
        {
          __typename: 'Attribute',
          displayValue: '1T',
          value: '1T',
          id: '1T',
        },
      ],
    },
  ],
  quantity: 1,
};

beforeEach(() => {
  wrapper = render(
    <BrowserRouter>
      <Provider store={store}>
        <CartItem cartItem={cartItem} />
      </Provider>
    </BrowserRouter>,
  );
});

describe('', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
