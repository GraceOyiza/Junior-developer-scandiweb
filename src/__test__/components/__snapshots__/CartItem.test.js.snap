// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders correctly 1`] = `
LoadedCheerio {
  "0": <div>
    <div
      class="sc-eDvSVe aNsuN"
    >
      <div>
        <div
          class="sc-gKPRtg jxKmWA"
        >
           PlayStation 5
        </div>
        <div
          class="sc-gKPRtg jxKmWA"
        >
           Sony
        </div>
        <div
          class="sc-iBYQkv jrCxzV"
        >
          $844.02
        </div>
        <div>
          <div>
            <div
              class="sc-gswNZR RkbKq"
            >
              Color:
            </div>
            <ul
              class="sc-dkrFOg jGHgsH"
            >
              <li
                class="sc-bcXHqe broeLV"
              >
                <input
                  id="color-0"
                  name="color"
                  type="radio"
                  value="#44FF03"
                />
                <label
                  class=" undefined swatch"
                  for="color-0"
                  style="background-color:#44FF03"
                >
                   
                </label>
              </li>
              <li
                class="sc-bcXHqe broeLV"
              >
                <input
                  id="color-1"
                  name="color"
                  type="radio"
                  value="#03FFF7"
                />
                <label
                  class=" undefined swatch"
                  for="color-1"
                  style="background-color:#03FFF7"
                >
                   
                </label>
              </li>
              <li
                class="sc-bcXHqe broeLV"
              >
                <input
                  id="color-2"
                  name="color"
                  type="radio"
                  value="#030BFF"
                />
                <label
                  class=" undefined swatch"
                  for="color-2"
                  style="background-color:#030BFF"
                >
                   
                </label>
              </li>
              <li
                class="sc-bcXHqe broeLV"
              >
                <input
                  id="color-3"
                  name="color"
                  type="radio"
                  value="#000000"
                />
                <label
                  class=" undefined swatch"
                  for="color-3"
                  style="background-color:#000000"
                >
                   
                </label>
              </li>
              <li
                class="sc-bcXHqe broeLV"
              >
                <input
                  id="color-4"
                  name="color"
                  type="radio"
                  value="#FFFFFF"
                />
                <label
                  class=" undefined swatch"
                  for="color-4"
                  style="background-color:#FFFFFF"
                >
                   
                </label>
              </li>
            </ul>
          </div>
        </div>
        <div>
          <div>
            <div
              class="sc-gswNZR RkbKq"
            >
              Capacity:
            </div>
            <ul
              class="sc-dkrFOg jGHgsH"
            >
              <li
                class="sc-bcXHqe broeLV"
              >
                <input
                  id="capacity-0"
                  name="capacity"
                  type="radio"
                  value="512G"
                />
                <label
                  class=" undefined false"
                  for="capacity-0"
                  style="background-color:"
                >
                  512G 
                </label>
              </li>
              <li
                class="sc-bcXHqe broeLV"
              >
                <input
                  id="capacity-1"
                  name="capacity"
                  type="radio"
                  value="1T"
                />
                <label
                  class=" undefined false"
                  for="capacity-1"
                  style="background-color:"
                >
                  1T 
                </label>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="sc-eDvSVe aNsuN"
      >
        <div
          class="sc-jSUZER cxDqCf"
        >
          <button
            class="sc-jrcTuL lizvMu"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 5V19"
                stroke-width="#32323A"
                strokewidth-linejoin="round"
                strokewidth-width="2"
                strokewidthlinecap="round"
              />
              <path
                d="M5 12H19"
                stroke-width="#32323A"
                strokewidth-linejoin="round"
                strokewidth-width="2"
                strokewidthlinecap="round"
              />
            </svg>
          </button>
          <input
            class="sc-bqWxrE gsuIFQ"
            value="1"
          />
          <button
            class="sc-jrcTuL lizvMu"
          >
            <svg
              fill="none"
              height="2"
              viewBox="0 0 16 2"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 1H15"
                stroke-width="#32323A"
                strokewidth-linejoin="round"
                strokewidth-width="2"
                strokewidthlinecap="round"
              />
            </svg>
          </button>
        </div>
        <div>
          <img
            class="sc-ftTHYK FZeuW"
            src="https://images-na.ssl-images-amazon.com/images/I/510VSJ9mWDL._SL1262_.jpg"
          />
        </div>
      </div>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
